src/components/GraphicsMenu.tsx

```diff
@@ -9,91 +10,218 @@ import {
+  editGraphicsMethod: Function;
+  copyGraphicsMethod: Function;
 };
 type GraphicsMenuState = {
   showMenu: boolean;
   showDropdown: boolean;
   selectedMethod: string;
   selectedGroup: string;
-  firstSelection: boolean;
-  optionsChanged: boolean;
+  tempGroup: string;
+  enterName: boolean;
+  nameValue: string;
+  invalidName: boolean;
   plotReady: boolean;
 };
 
 export default class GraphicsMenu extends React.Component<
   GraphicsMenuProps,
   GraphicsMenuState
 > {
+  nameInputRef: Input;
   constructor(props: GraphicsMenuProps) {
     super(props);
     this.state = {
       showMenu: false,
       showDropdown: false,
       selectedMethod: "",
-      selectedGroup: "Select Plot Type",
-      firstSelection: false,
-      optionsChanged: false,
+      selectedGroup: "",
+      tempGroup: "",
+      enterName: false,
+      nameValue: "",
+      invalidName: false,
       plotReady: false
     };
-    this.toggleMenu = this.toggleMenu.bind(this);
     this.toggleDropdown = this.toggleDropdown.bind(this);
     this.graphicsOptions = this.graphicsOptions.bind(this);
-    this.selectFalse = this.selectFalse.bind(this);
-    this.selectTrue = this.selectTrue.bind(this);
+    this.resetGraphicsState = this.resetGraphicsState.bind(this);
+    this.handleNameInput = this.handleNameInput.bind(this);
+    this.selectItem = this.selectItem.bind(this);
+    this.nameInputRef = (React as any).createRef();
   }
-  toggleMenu() {
-    this.setState({
-      showMenu: !this.state.showMenu
-    });
+
+  handleNameInput(event: React.ChangeEvent<HTMLInputElement>) {
+    // Regex filter for unallowed name characters
+    let forbidden: RegExp = /^[^A-z_]|[^A-z0-9]+/;
+    let invalid: boolean = forbidden.test(event.target.value);
+    this.setState({ nameValue: event.target.value, invalidName: invalid });
   }
+
   toggleDropdown() {
     this.setState({
       showDropdown: !this.state.showDropdown
     });
+    if (this.state.showMenu && this.state.enterName) {
+      this.setState({
+        showMenu: false
+      });
+    }
   }
+
+  // Resets the graphics menu to initial, (for when a new notebook is selected)
+  resetGraphicsState() {
+    console.log("Graphics selections reset.");
+    this.setState({
+      showMenu: false,
+      showDropdown: false,
+      selectedMethod: "",
+      selectedGroup: "",
+      tempGroup: "",
+      enterName: false,
+      nameValue: "",
+      invalidName: false,
+      plotReady: false
+    });
+  }
+
+  async selectItem(item: string) {
+    try {
+      if (
+        this.state.tempGroup != this.state.selectedGroup ||
+        this.state.selectedMethod != item
+      ) {
+        await this.props.updateGraphicsOptions(this.state.tempGroup, item);
+
+        this.setState({
+          showMenu: false,
+          selectedMethod: item,
+          selectedGroup: this.state.tempGroup
+        });
+        console.log(`Updated graphics to: ${item}`);
```

Comment:
console logs are a great tool for debugging, but they should be removed before merging into a branch users will actually see.